### PR TITLE
added phpstan callable types

### PR DESCRIPTION
**Pull request by clxmstaab**
- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [x] Changelog updated?

      Please, update CHANGELOG.md by running next command:
      $ php bin/changelog

added types where callables are used.

see https://phpstan.org/writing-php-code/phpdoc-types#callables

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -227,11 +228,12 @@ function after(string $task, $do)
...
  *
  * @param string $task The task which need to fail so $that should be run.
- * @param string $do The task to be run.
+ * @param string|callable $do The task to be run.
+ * @phpstan-param string|callable(): void $do
  *
  * @return Task|void
  */
-function fail(string $task, string $do)
+function fail(string $task, $do)
```
Comment (clxmstaab) [inline src/functions.php:236]: in this line a bug was fixed. `fail` also accepts a callable, and therefore the native type was wrong.

File: src/Task/Task.php [hunk 2]
```diff
@@ -12,6 +12,9 @@
 class Task
 {
     private $name;
+    /**
+     * @phpstan-var callable():void $callback
```
Comment (clxmstaab) [inline src/Task/Task.php:16]:
```suggestion
     * @phpstan-var callable():void
```

File: src/functions.php (hunk 1 above)
Comment (clxmstaab) [reply src/functions.php:236]: should I add a changelog entry for this fix?

**Comment by clxmstaab** [comment]
conflicts resolved

**Comment by clxmstaab** [comment]
created a upstream issue regarding the remaining phpstan errors:

image


https://github.com/phpstan/phpstan/issues/4046

**Comment by clxmstaab** [comment]
extracted the bugfix into https://github.com/deployphp/deployer/pull/2261 since this callable signature PR is blocked on the upstream issue https://github.com/phpstan/phpstan/issues/3115

**Comment by antonmedv** [comment]
I'm going to close this PR for cleaning purposes. Feel free to reopen this PR if needed.

**Pull request closed without merging**